lib/Onyx.js

```diff
@@ -1222,6 +1222,9 @@ function update(data) {
  */
 function setMemoryOnlyKeys(keyList) {
     Storage.setMemoryOnlyKeys(keyList);
+
+    // When in memory only mode for certain keys we do not want to ever drop items from the cache as the user will have no way to recover them again via storage.
+    cache.setRecentKeysLimit(Infinity);
```

Comment:
This is basically a round about way to opt out of the logic that runs here:

https://github.com/Expensify/react-native-onyx/blob/31083216bb8d29dd114f4efe9ef041fe8fedfb5a/lib/OnyxCache.js#L181-L192